tools/interop_matrix/client_matrix.py

```diff
@@ -110,52 +125,89 @@ def should_build_docker_interop_image_from_release_tag(lang):
     ],
     'go': [
         {
-            'v1.0.5': None
+            'v1.0.5': {
+                'skip_runtime': ['go1.11']
```

Comment:
Could this be inverted so you specify runtimes used instead of skipped?  Otherwise this list will need to continually grow as we add runtimes.

Reply:
That's a good point. I might take a look into that in a follow-up pull request (I have some run_interop_matrix_tests.py simplification in progress, so I will see if I can easily add to that PR).
In this PR, I prefer to leave as is as it solves the problem, we rarely add new runtimes, and I don't want to further delay fixing #16928 and adding grpc-go 1.17 interop coverage.

Reply:
> we rarely add new runtimes

What is the purpose of adding runtimes to this framework?  Is the goal to have *some* runtime for every release of grpc-go, or are we trying to test *all* grpc-go versions with *all* runtimes supported by them?  If the latter, we are missing go1.9 and go1.10, and a new Go release happens every 6 months.  If the former, we might as well remove go1.7 -- or, we could delete the concept of runtimes from this script entirely, and when we generate an image for a grpc-go release, we will just make sure the image uses a runtime that works with it.

Reply:
I'd say choosing the runtimes to use is up to the language owners - Ideally we would test everything, but than the test suite would take very long to run, thus limiting the usefulness. So it's a tradeoff between the number of runtimes to test vs infrastructure complexity and time to run.
Ad "removing the concept of runtimes entirely": I'm totally fine with removing some runtimes  if you think that's a good choice for grpc-go (leaving up to you). But there still needs to be some logic that decides which dockerfiles/runtimes will be used with which release tag (right now that's declared explicitly in the data structures in client_matrix.py, which makes sense IMHO). So even if we only test one runtime per release tag, there will still be a concept of runtime in the interop_matrix scripts (so this wouldn't really simplify the test scripts much).

Are you ok with me merging this PR?

Reply:
For Go, I don't think there's much value in testing multiple runtimes per release.  As long as we have one that works for every grpc-go version, we should be fine.  There are not substantial differences across Go releases, and they are supposed to be ~100% backward compatible.

> Are you ok with me merging this PR?

Yes, since it unblocks testing grpc-go 1.17 in interop.  But please invert the way this works to specify good grpc+runtime combinations in a follow-up soon to avoid the long-term maintenance costs as the number of runtimes and releases grow.

Reply:
Sg.